pkg/component/gardener/dashboard/terminal/deployment.go

```diff
@@ -0,0 +1,183 @@
+						v1beta1constants.LabelNetworkPolicyToRuntimeAPIServer: v1beta1constants.LabelNetworkPolicyAllowed,
+						v1beta1constants.LabelNetworkPolicyToPublicNetworks:   v1beta1constants.LabelNetworkPolicyAllowed,
+						v1beta1constants.LabelNetworkPolicyToPrivateNetworks:  v1beta1constants.LabelNetworkPolicyAllowed,
+						gardenerutils.NetworkPolicyLabel("virtual-garden-"+v1beta1constants.DeploymentNameKubeAPIServer, kubeapiserverconstants.Port): v1beta1constants.LabelNetworkPolicyAllowed,
+					}),
+				},
+				Spec: corev1.PodSpec{
+					PriorityClassName:            v1beta1constants.PriorityClassNameGardenSystem200,
+					AutomountServiceAccountToken: ptr.To(false),
+					SecurityContext: &corev1.PodSecurityContext{
+						RunAsNonRoot: ptr.To(true),
+						RunAsUser:    ptr.To[int64](65532),
+					},
+					TerminationGracePeriodSeconds: ptr.To(int64(10)),
+					Containers: []corev1.Container{{
+						Name:            name,
+						Image:           t.values.Image,
+						ImagePullPolicy: corev1.PullIfNotPresent,
+						Args: []string{
+							"--config-file=" + volumeMountPathConfig + "/" + dataKeyConfig,
+							"--zap-devel=false",
+						},
+						Env: []corev1.EnvVar{{
+							Name:  "KUBECONFIG",
+							Value: gardenerutils.PathGenericKubeconfig,
+						}},
+						Resources: corev1.ResourceRequirements{
+							Requests: map[corev1.ResourceName]resource.Quantity{
+								corev1.ResourceCPU:    resource.MustParse("10m"),
+								corev1.ResourceMemory: resource.MustParse("50Mi"),
+							},
+						},
+						Ports: []corev1.ContainerPort{
+							{
+								Name:          portNameAdmission,
+								ContainerPort: portAdmission,
+								Protocol:      corev1.ProtocolTCP,
+							},
+							{
+								Name:          portNameMetrics,
+								ContainerPort: portMetrics,
+								Protocol:      corev1.ProtocolTCP,
+							},
+						},
+						LivenessProbe: &corev1.Probe{
+							ProbeHandler: corev1.ProbeHandler{
+								HTTPGet: &corev1.HTTPGetAction{
+									Path:   "/healthz",
+									Port:   intstr.FromInt32(portProbes),
+									Scheme: corev1.URISchemeHTTP,
+								},
+							},
+							InitialDelaySeconds: 15,
+							TimeoutSeconds:      5,
+							FailureThreshold:    6,
+							SuccessThreshold:    1,
+							PeriodSeconds:       20,
+						},
+						ReadinessProbe: &corev1.Probe{
+							ProbeHandler: corev1.ProbeHandler{
+								HTTPGet: &corev1.HTTPGetAction{
+									Path:   "/readyz",
+									Port:   intstr.FromInt32(portProbes),
+									Scheme: corev1.URISchemeHTTP,
+								},
+							},
+							InitialDelaySeconds: 5,
+							TimeoutSeconds:      5,
+							FailureThreshold:    6,
+							SuccessThreshold:    1,
+							PeriodSeconds:       10,
+						},
+						SecurityContext: &corev1.SecurityContext{AllowPrivilegeEscalation: ptr.To(false)},
+						VolumeMounts: []corev1.VolumeMount{
+							{
+								Name:      volumeNameConfig,
+								MountPath: volumeMountPathConfig,
+							},
```

Comment:
```suggestion
								MountPath: volumeMountPathConfig,
								ReadOnly:  true,
							},
```
will also be changed in TCM helm chart accordingly https://github.com/gardener/terminal-controller-manager/pull/289